lib/index.js

```diff
@@ -290,7 +326,7 @@ util.inherits(Device, EventEmitter);
  */
 function createSocket (type, opts) {
     var domain = (opts || {}).raw ? nn.AF_SP_RAW : nn.AF_SP;
-    return new Socket(domain, type);
+    return new Socket(domain, type, opts);
```

Comment:
ya i was doing that but for some reason the `transform.js` test would time out.. tape would just stall forever wtf so i just left domain untouched. 

I'll restart my machine and pull down a fresh copy, not sure why domain opt would have anything to do with that test timing out..